raiden_contracts/tests/utils.py

```diff
@@ -156,6 +156,87 @@ def get_settlement_amounts(
         participant2_locked=participant2_locked,
     )
 
+def get_onchain_settlement_amounts(
+        participant1,
+        participant2
+):
+    """ Settlement algorithm
+
+    Calculates the token amounts to be transferred to the channel participants when
+    a channel is settled.
+
+    !!! Don't change this unless you really know what you are doing.
+    """
+
+    def failsafe_add(a, b):
```

Comment:
Would be nice to have these helper functions outside the `get_onchain_settlement_amounts` function, so we can use them to test the internal `failsafe_add` function in the smart contract separately.